histomicstk/FeatureExtractionHaralick.py

```diff
@@ -0,0 +1,106 @@
+import skimage.feature
+import numpy as np
+
+
+def FeatureExtractionHaralick(I, Dst=1, Ng=256):
+    """
+    Calculates 26 Haralick features from an image
+    Parameters
+    ----------
+    I : array_like
+        An intensity image
+    Dst : Distance of neighboring pixels in an image. Default value = 1.
+    Ng : Number of gray level co-occurance matrix. Default value = 256.
+    Returns
+    -------
+    H : array_like
+        A matrix containing mean and ranges for 26 features
+    References
+    ----------
+    .. [1] Haralick, et al. "Textural features for image classification,"
+    IEEE Transactions on Systems, Man, and Cybernatics, vol. 6, pp: 610-621,
+    1973.
+    .. [2] Luis Pedro Coelho. "Mahotas: Open source software for scriptable
+    computer vision," Journal of Open Research Software, vol 1, 2013.
+    """
+    # sets the number of angles: 0 45 90 135. this must not be modified.
+    NumberofAngles = 4
```

Comment:
@slee172 I was thinking people might want to do it just for horizontal and vertical adjacency. 

Also, the names of each of the features is not being returned which i think is important if we want to figure out which feature is important for a classification task or which feature correlates strongly with some genetic variable. Maybe return a pandas data frame.

Also, i see that these features are being computed for the whole image. We might also want to apply this for tissue/region classification and cell classification. So, I think it would be more generic if it takes an image and binary mask as input and compute the GLCM matrix for only those pairs of neighboring pixels that are within the mask (maybe with atleast one of the pixels in the mask). After that, you can compute all the features derived out of the GLCM matrix that you are currently computing.

CC: @cooperlab